.github/scripts/sync_labels.py

```diff
@@ -48,12 +48,14 @@ def get_linked_pr_from_issue_number(repo, number):
 
 def get_linked_issues_based_on_pr_body(repo, number):
     pr = repo.get_pull(number)
-    pattern = fr'Fixes:? (?:#|{repo}#|https://github.com/{repo}/issues/)(\d+)'
-    matches = re.findall(pattern, pr.body)
+    pattern = fr'(fix(?:|es|ed)?|resolve(?:|d|s)?)?(?:#|{repo}?:#(\d+)|https.*?/issues/)(\d+)'
```

Comment:
one, sorry, but I think that `(?:|es|ed)?` is extraneous since the first `|` means an empty string as one of the options, so the `?` isn't really needed.l (I missed this subtlety in the previous version).

I still don't understand the second part though.
Have you tested it?
It'd be good to have a test function with this regex that would run it against a series of typical test strings we consider legal.  why is there `\d+` both inside the parenthsis and outside?
also if you want to put the issue number in a group with parenthesis `(\d+)` better use a named group, otherwise I don't think the parenthesis is needed.
In addition how about whitespace between the first part (fix/resolve/close) and the second?

Also, a patch may close a PR nit only fix/resolve an issue, or is this expression targeting only issued and PRs?

Reply:
So we should support the following options:
```
fix: #22
closes scylladb/scylla#55
resolved https://github.com/issues/1234
```

Modified the pattern with the following:
- added back `close|d|s`
- remove the extra `(\d+)` (you are right, it's not needed)
- removed the parenthesis at the end so we have now `\d+`

Reply:
Seems i still missing something , will test it again and ping you when ready

Reply:
@bhalevy Fixed, not all mention syntax are working as expected